recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.py

```diff
@@ -0,0 +1,193 @@
+        if current_id not in cpu_device_id:
+            FirmwareUpdateException("Upgrade is not supported for the board")
+
+    def update_firmware(self, filename):
+        """Update Firmware"""
+        mtd_num = 0
+        img_pos = 0
+        skip_bytes = 0
+
+        print("===================================================")
+        print("IOT2050 firmware update started - DO NOT INTERRUPT!")
+        print("===================================================")
+
+        while True:
+            mtd_sys_path = "/sys/bus/platform/devices/47040000.spi/mtd/mtd{}".format(mtd_num)
+            mtd_name_path = "{}/name".format(mtd_sys_path)
+            mtd_size_path = "{}/size".format(mtd_sys_path)
+            mtd_dev_path = "/dev/mtd{}".format(mtd_num)
+
+            mtd_size = self.get_path_type_value(mtd_size_path)
+            mtd_size = int(mtd_size)
+
+            mtd_name = self.get_path_type_value(mtd_name_path)
+            env_list = ["ospi.env","ospi.env.backup"]
+
+            if not self.erase_env and mtd_name.strip() in env_list:
+                skip_bytes = skip_bytes+mtd_size
+                mtd_num = mtd_num+1
+                print("Skipping update %-20s" % mtd_name)
+                continue
+            else:
+                print("Updating %-20s" % mtd_name.strip(),end="")
+
+            if mtd_name.strip() == "ospi.rootfs":
+                print("\n\nCompleted. Please reboot the device\n")
+                return
+            try:
+                with open(mtd_dev_path, 'rb') as mtd:
+                    mtd_contents = mtd.read()
+            except FileNotFoundError:
+                print("Open %s failed" % mtd_dev_path)
+                sys.exit(1)
+
+            with open(filename, 'rb') as firmware:
+                firmware.seek(img_pos+skip_bytes)
+                firmware_contents = firmware.read(mtd_size)
+
+            if not mtd_contents == firmware_contents:
+                print("U")
+                self.flash_erase(mtd_dev_path, 0, mtd_size)
+                with open(mtd_dev_path, 'wb') as mtd_unity:
+                    mtd_unity.write(firmware_contents)
+            else:
+                print(".")
+            img_pos = img_pos+mtd_size
+            mtd_num = (mtd_num+1)
+            print("\n")
+
+
+def main(argv):
+    parser = OptionParser()
+    parser.add_option('-f', '--force', action='store_true', dest='force', default=False, help="Force update the firmware,ignore the error")
+    parser.add_option('-e', '--erase', action='store_true', dest='erase', default=False, help="Erase the Environment")
+
+    (options, args) = parser.parse_args()
+
+    if options.force:
+        result = input("\nDo force update.Maybe Destory your System:Y/n ")
+        if result == "n":
+            options.force = False
+        else:
+            global force_update
+            force_update = options.force
+
+    if not args:
+        print("Require Firmware")
+        sys.exit(1)
```

Comment:
Flash file must be modeled as positional parameter so that --help also shows it. I'm using ArgumentParser elsewhere, but I suspect the OptionParser can do that as well.